src/main/scala/org/apache/spark/sql/kinesis/KinesisReader.scala

```diff
@@ -170,36 +168,18 @@ private[kinesis] case class KinesisReader(
     records
   }
 
-  private def describeKinesisStream(): Seq[Shard] = {
-    // TODO - We have a limit on DescribeStream API call.
-    // So we should be cautious before making this call
+  private def listShards(): Seq[Shard] = {
 
-    val describeStreamRequest = new DescribeStreamRequest
-    describeStreamRequest.setStreamName(streamName)
-    describeStreamRequest.setLimit(maxSupportedShardsPerStream)
+    val listShardsRequest = new ListShardsRequest
+    listShardsRequest.setStreamName(streamName)
 
-    val describeStreamResult: DescribeStreamResult = runUninterruptibly {
-      retryOrTimeout[DescribeStreamResult]( s"Describe Streams") {
-          getAmazonClient.describeStream(describeStreamRequest)
+    val listShardsResult: ListShardsResult = runUninterruptibly {
+      retryOrTimeout[ListShardsResult]( s"List shards") {
+          getAmazonClient.listShards(listShardsRequest)
```

Comment:
maximum Result limit is 100 (https://docs.aws.amazon.com/kinesis/latest/APIReference/API_ListShards.html#Streams-ListShards-request-MaxResults)  if we have more than 100 shards in the stream, we would need to use nextToken to get more shards

or is there anyway to increase the maxResult Limit

Reply:
Good call